cpp/src/arrow/compute/row/compare_test.cc

```diff
@@ -166,10 +166,9 @@ TEST(KeyCompare, CompareColumnsToRowsTempStackUsage) {
   }
 }
 
-#ifndef ARROW_VALGRIND
```

Comment:
I guess we don't care much about the execution time we are running "large memory tests", even with Valgrind, so removing it.